52.md

```diff
@@ -33,6 +33,28 @@ The following tags are deprecated:
 
 Calendar events are _not_ required to be part of a [calendar](#calendar).
 
+### Private events
+Private calendar events are events between users which only the specifically invited users can see and respond to. It is not necessary that users who are invited may or may not be participants in that event. Private calendar events follow a similar approach to the corresponding public calendar events but they also include a `description` tag which will contain the description of the event. This is necessary as the content field will contain the complete encrypted event itself.
+
+Please note the the below implementation is meant to be a generic implementation for private events as well as RSVPs with minor differences. The specific kinds of the encrypted event and the gift wraps are defined in the specific sections
+
+#### Publishing Private events
+
+1. Generate a private key. We will call this a `viewKey`. This key will be used to encrypt the event and will be sent to the viewers
+2. Encrypt all the tags of the event with the `viewKey` and put the encrypted string in the content field of the final event.
+3. Only include in the `d` tag in the final event. We will call the the unique calendar event id  as `encrypted_event_id`
+4. Now gift wrap(nip 59) the `viewKey` and the `encrypted_event_id` in a rumor. The `.content` field of the rumor will be an empty string. The rumor will contain the following tags
+    1. `a` (required). Value should be `<encrypted_event_kind>:<viewer_pubkey>:<encrypted_event_id>`
+    2. `viewkey` (required). Value should be `viewKey`
+5. The gift wrap itself should have a `p` tag with the viewer pubKey.
+6. Separate gift wraps should be created for each viewer.
```

Comment:
If you are encrypting the wrap to each user, why do you need the shared viewkey? You can just encrypt everything to each receiving user. There is no need to manage this separate key.

Reply:
I think it's because you need the ability to edit the previous event, something you lose if you just giftwrap the entire thing, or at-least you deal with a huge network overhead.

Reply:
It will make editing the original event much easier. Once the wrap with the viewkey is sent, the participants will always have access to the event. Editing the original event/adding new participants will be very simple. It can be thought of as a 2 step process:
1. Create an event
2. Give access to the event

This way, the access management can also scale in the future. There can be different access levels like read/write etc(although this nip enhancement does not cover those scenarios).